### PR TITLE
Update kotlin-metadata-jvm

### PR DESCRIPTION
**Pull request by fzhinkin**
Replaced `kotlinx-metadata-jvm` with `kotlin-metadata-jvm,` and updated the code to use a new API.

### CODE REVIEW

File: src/main/kotlin/api/KotlinSignaturesLoading.kt [hunk 1]
```diff
@@ -132,10 +131,9 @@ private fun FieldNode.buildFieldSignature(
         }
 
         val property = companionClassCandidate?.kmProperty(name)
-
-        if (property != null && JvmFlag.Property.IS_MOVED_FROM_INTERFACE_COMPANION(property.flags)) {
```
Comment (fzhinkin) [inline src/main/kotlin/api/KotlinSignaturesLoading.kt:136]: That was a bug, and `JvmFlag.Property.IS_MOVED_FROM_INTERFACE_COMPANION` worked here purely by coincidence.

File: src/main/kotlin/api/KotlinMetadataVisibilities.kt [hunk 2]
```diff
@@ -68,15 +67,7 @@ internal val ClassNode.kotlinMetadata: KotlinClassMetadata?
...
-        return KotlinClassMetadata.read(header)
-            ?: error(
-                """
-                Incompatible version of Kotlin metadata.
-                Maximal supported Kotlin metadata version: ${COMPATIBLE_METADATA_VERSION.joinToString(".")},
-                $name Kotlin metadata version: ${header.metadataVersion.joinToString(".")}.
-                As a workaround, it is possible to manually update 'kotlinx-metadata-jvm' version in your project.
-            """.trimIndent()
-            )
+        return KotlinClassMetadata.readStrict(header)
```
Comment (sandwwraith) [inline src/main/kotlin/api/KotlinMetadataVisibilities.kt:70]: I think it is possible here to use `readLenient()` — we added this function specifically with BCV in mind. See the difference explained here: https://github.com/JetBrains/kotlin/blob/master/libraries/kotlinx-metadata/jvm/ReadMe.md#working-with-different-versions

Comment (fzhinkin) [reply src/main/kotlin/api/KotlinMetadataVisibilities.kt:70]: Yeah, it makes perfect sense

File: src/main/kotlin/api/KotlinSignaturesLoading.kt (hunk 1 above)
Comment (fzhinkin) [reply src/main/kotlin/api/KotlinSignaturesLoading.kt:136]: Thanks to the new stable API, it became obvious! ;)

**Review by shanshin — APPROVED** [review]

**Comment by fzhinkin** [comment]
kotlin-metadata-jvm seems to be API-compatible with Kotlin 1.6, but the API version is not set explicitly. Here's an issue on that: https://youtrack.jetbrains.com/issue/KT-69903

**Pull request merged**